packages/integration/api-report.md

```diff
@@ -4,8 +4,62 @@
 
 ```ts
 import { Config } from '@backstage/config';
+import { CredentialProvider } from '@aws-sdk/types';
```

Comment:
I'm slightly worried about exposing this types from the AWS SDK our public API in case aws-sdk introduce some breaking changes going forward. Would it be feasible to return our own provider somehow?

Reply:
Hmmmm I think that would be tricky because AWS credentials can be temporary and expire, so it's not as straightforward as just returning static credentials.  If we simply return temporary credentials in our own provider, each plugin using the provider needs to correctly handle credential expiration, which is likely to be error-prone.

The AWS SDK clients don't take credentials directly as params; they take a function that gets called each time the client makes an API call.  For temporary creds, that function is responsible for handling credential expiration and refreshing the creds as needed; it is not the direct responsibility of the client to manage expiry.
```typescript
    const creds = await awsCredentialsProvider.getCredentials({ arn });
    const client = new ProtonClient({
      region,
      // creds.provider is a function that returns Promise<Credentials>
      credentialDefaultProvider: () => creds.provider,
    });
```